gemfirexd/tools/src/main/java/com/pivotal/gemfirexd/tools/internal/GfxdServerLauncher.java

```diff
@@ -494,6 +495,10 @@ else if (WAIT_FOR_SYNC.equals(key)) {
       }
       this.waitForData = "true".equalsIgnoreCase(value);
     }
+    else if (MEMORY_SIZE.equals(key)) {
+      props.setProperty(DistributionConfig.MEMORY_SIZE_NAME, value);
+      this.memorySize = value;
+    }
```

Comment:
Why is this special handling required? Since the name in DistributionConfig is the same, all properties on command-line are put into the distribution props already so changes to GfxdServerLauncher can be removed.